referenceqvm/unitary_generator.py

```diff
@@ -355,6 +356,45 @@ def tensor_gates(gate_set, defgate_set, pyquil_gate, num_qubits):
     return gate
 
 
+def tensor_up(pauli_terms, num_qubits):
+    """
+    Takes a PauliSum object along with a total number of
+    qubits and returns a matrix corresponding the tensor representation of the
+    object.
+
+    Useful for generating the full Hamiltonian after a particular fermion to
+    pauli transformation.
+
+    :param pauli_terms: (PauliSum) object of PauliTerm
+    :param num_qubits: (int) number of qubits in the system
+    :returns: (numpy array) representation of the paui_terms operator
+    """
+    if not isinstance(pauli_terms, PauliSum):
+        raise TypeError("can only tensor PauliSum")
+
+    if __debug__:
+        for term in pauli_terms.terms:
+            if len(term._ops.keys()) > 0:
+                if max(term._ops.keys()) >= num_qubits:
+                    raise IndexError("pauli_terms has higher index than qubits")
+
+    big_hilbert = np.zeros((2 ** num_qubits, 2 ** num_qubits))
+    # left kronecker product corresponds to the correct basis ordering
+    for term in pauli_terms.terms:
+        tmp_big_hilbert = np.array([1])
+
+        for index in range(num_qubits):
+            pauli_mat = gate_matrix[term[index]]
+
+            tmp_big_hilbert = np.kron(pauli_mat, tmp_big_hilbert)
```

Comment:
Collapse this in one line. The additional assignment does not increase readability.
```
tmp_big_hilbert = np.kron(gate_matrix[term[index]], tmp_big_hilbert)
```